src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -138,45 +138,50 @@ private void EmbedIfNeedTo(BoundExpression receiver, ImmutableArray<PropertySymb
             ref BoundExpression? receiverOpt,
             ref ImmutableArray<BoundExpression> arguments,
             ref ImmutableArray<RefKind> argumentRefKindsOpt,
-            ref bool invokedAsExtensionMethod,
             Location? interceptableLocation)
         {
             // PROTOTYPE(ic):
             // Add assertions for the possible shapes of calls which could come through this method.
             // When the BoundCall shape changes in the future, force developer to decide what to do here.
 
-            // PROTOTYPE: perhaps a 'TryGet' pattern is more suitable here.
-            if (this._compilation.GetInterceptor(interceptableLocation) is not var (interceptsLocationAttributeData, interceptor))
+            if (this._compilation.TryGetInterceptor(interceptableLocation) is not var (attributeLocation, interceptor))
```

Comment:
I was proposing something like `bool TryGetInterceptor TryGetInterceptor(Location? callLocation, out Location AttributeLocation, out MethodSymbol Interceptor)`. 
Then we can just do `if (this._compilation.TryGetInterceptor(interceptableLocation, out var attributeLocation, out var interceptor))` without a positional pattern.

Reply:
To me, this feels comparable in stack utilization and ergonomics to the conventional `TryGet` pattern, while being easier on the implementation side:
1. don't need `[NotNullWhen]`
2. don't need to assign out parameters potentially in multiple return paths, or risk forgetting to assign them.

I could be missing something, though, or under-valuing the benefit of adhering to convention.